app/javascript/components/dual-list-select/index.jsx

```diff
@@ -0,0 +1,165 @@
+        <div className="col-md-5">
+          {leftTitle}
+          <List size={size} ref={this.leftList} id={leftId} values={leftValues} />
+        </div>
+        <div className="col-md-1" style={{ padding: 10 }}>
+          <div className="spacer" />
+          <div className="spacer" />
+          <Button disabled={isEmpty(leftValues)} className="btn-block" onClick={this.moveRight} title={moveRightTitle}>
+            <Icon type="fa" name="angle-right" size="lg" />
+          </Button>
+          { allToRight
+          && (
+          <Button disabled={isEmpty(leftValues)} className="btn-block" onClick={() => this.moveAllRight(leftValues)} title={moveAllRightTitle}>
+            <Icon type="fa" name="angle-double-right" size="lg" />
+          </Button>
+          ) }
+          { allToLeft
+          && (
+          <Button disabled={isEmpty(rightValues)} className="btn-block" onClick={this.moveAllLeft} title={moveAllLeftTitle}>
+            <Icon type="fa" name="angle-double-left" size="lg" />
+          </Button>
+          ) }
+          <Button disabled={isEmpty(rightValues)} className="btn-block" onClick={() => this.moveLeft(rightValues)} title={moveLeftTitle}>
+            <Icon type="fa" name="angle-left" size="lg" />
+          </Button>
+        </div>
+        <div className="col-md-5">
+          {rightTitle}
+          <List size={size} ref={this.rightList} id={rightId} values={rightValues} />
+        </div>
+      </div>);
+  }
+}
+
+DualListSelect.propTypes = {
+  /* ID of the left select */
+  leftId: PropTypes.string,
+  /* ID of the right select */
+  rightId: PropTypes.string,
+  /* Title above left select */
+  leftTitle: PropTypes.string,
+  /* Title above right select */
+  rightTitle: PropTypes.string,
+  /* Size of select */
+  size: PropTypes.number,
+  /* Should show button "move all to right" */
+  allToRight: PropTypes.bool,
+  /* Should show button "move all to left" */
+  allToLeft: PropTypes.bool,
+  /* Title of move left button */
+  moveLeftTitle: PropTypes.string,
+  /* Title of move right button */
+  moveRightTitle: PropTypes.string,
+  /* Title of move all right button */
+  moveAllRightTitle: PropTypes.string,
+  /* Title of move all left button */
+  moveAllLeftTitle: PropTypes.string,
+  /* Values in right select { value: "Text", value1: "Text1", ... } */
+  options: PropTypes.shape({
+    [PropTypes.string]: PropTypes.string,
+  }),
+};
+
+DualListSelect.defaultProps = {
+  leftId: undefined,
+  rightId: undefined,
+  leftTitle: undefined,
+  rightTitle: undefined,
+  size: 15,
+  allToRight: true,
+  allToLeft: false,
+  moveLeftTitle: __('Move selected to left'),
+  moveRightTitle: __('Move selected to right'),
+  moveAllRightTitle: __('Move all to right'),
+  moveAllLeftTitle: __('Move all to left'),
+  options: {},
+};
+
+const WrappedDualList = ({ FieldProvider, name, ...rest }) => (
+  <FieldProvider
+    name={name}
+    {...rest}
+    component={props => <DualListSelect {...props} />}
```

Comment:
I you passing all props without any changes, you can pass just the component reference:
```JSX
component={DualListSelect}
```

Reply:
It throws `Uncaught Error: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: object.` :sob:

Reply:
ok nvm